src/transformers/models/gptj/modeling_gptj.py

```diff
@@ -233,8 +233,8 @@ def forward(
             key = apply_rotary_pos_emb(key, sin, cos)
             query = apply_rotary_pos_emb(query, sin, cos)
 
-        key = key.permute(0, 2, 1, 3)
-        query = query.permute(0, 2, 1, 3)
+        key = key.permute(0, 2, 1, 3).to(value.dtype)
+        query = query.permute(0, 2, 1, 3).to(value.dtype)
```

Comment:
could you explain why this is needed for this particular model and why this doesn't happen for llama for example ? Many models have approximately the same modeling code.

Reply:
For llama, we can see sin and cos come from position_embeddings (bf16 tensor) which comes from [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L168). You can see llama's rotary embedding converts the data type. But for gptj, the position embeddings come from [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/gptj/modeling_gptj.py#L63), it set the data type to float32, so data type miss-match happens when the input data type is bf16 or fp16.